Gems/Terrain/Code/Source/Components/TerrainWorldDebuggerComponent.cpp

```diff
@@ -282,16 +282,21 @@ namespace Terrain
                     AZ::Vector3(sectorX * sectorSize.GetX(), sectorY * sectorSize.GetY(), worldMinZ),
                     AZ::Vector3((sectorX + 1) * sectorSize.GetX(), (sectorY + 1) * sectorSize.GetY(), worldMinZ));
 
-                // Clamp it to the terrain world bounds.
-                sectorAabb.Clamp(worldBounds);
-
                 // If the world space box for the sector doesn't match, set it and mark the sector as dirty so we refresh the height data.
                 {
                     AZStd::lock_guard<AZStd::recursive_mutex> lock(sector.m_sectorStateMutex);
                     if (sector.m_aabb != sectorAabb)
                     {
                         sector.m_aabb = sectorAabb;
```

Comment:
If I moved it, the "if" check would fail on every frame even when nothing is moving, causing it to check the Overlaps() on every frame, so I'll leave it where it is.